Tasks/Common/MSBuildHelpers/msbuildhelpers.ts

```diff
@@ -0,0 +1,50 @@
+import tl = require('vsts-task-lib/task');
+import { ToolRunner } from 'vsts-task-lib/toolrunner';
+
+/**
+ * Finds the tool path for msbuild/xbuild based on specified msbuild version on Mac or Linux agent
+ * @param version 
+ */
+export async function getMSBuildPath(version) {
+    let toolPath: string;
+
+    if (version === '15.0' || version === 'latest') {
+        let msbuildPath: string = tl.which('msbuild', false);
```

Comment:
This logic is different to how vsts-agent detects the correct MSBuild - https://github.com/Microsoft/vsts-agent/commit/8f25b5bffae36594f601e2c178c88c6dcc675c3f

Reply:
@alanmcgovern : I think the way the agent is detecting the capability for msbuild is incorrect. It should pick the one in the path and look under a specific location which may or may not apply to all macOS machines. Are you using the Xamarin.iOS task or msbuild task to build on macOS. Please share the task settings and version.